### DISCUSSION TITLE
Source download failing after upgrading terragrunt to 0.55.6

### DISCUSSION THREAD

**Discussion by anupchandak-ibm** [gruntwork-io/knowledge-base#844] [Help]
Hi Team 

We were using Terragrunt 0.53 where below block was working and was able to download the module from Artifactory registry 

```
terraform {
  source = "https://my.artifactory-devops.com/artifactory/my-custom-modules-terraform-local/aws/my-vpc/1.0.0.zip"
}
```

But after upgrading Terragrunt to 0.55.6, its trying to download it as local file and hence failing with below error 

```
anupchandak@Anups-MacBook-Pro mcsp-createvpc-main % TG_LOG=debug terragrunt init
INFO[0000] Downloading Terraform configurations from file:///Users/anupchandak/git/my-terraform-inventory/aws-rel/us03/us03a/mcsp-createvpc-main/my.artifactory-devops.com/artifactory/my-custom-modules-terraform-local/aws/my-vpc/1.0.0.zip into /Users/anupchandak/git/my-terraform-inventory/aws-rel/us03/us03a/mcsp-createvpc-main/.terragrunt-cache/OXhG88jWiWQI5clPG9ebpzKmCNA/_43skhcXB65TEpVs_JG_OjXI3TU 
ERRO[0000] downloading source url file:///Users/anupchandak/git/my-terraform-inventory/aws-rel/us03/us03a/mcsp-createvpc-main/my.artifactory-devops.com/artifactory/my-custom-modules-terraform-local/aws/my-vpc/1.0.0.zip
1 error occurred:
        * source path error: stat /Users/anupchandak/git/my-terraform-inventory/aws-rel/us03/us03a/mcsp-createvpc-main/my.artifactory-devops.com/artifactory/my-custom-modules-terraform-local/aws/my-vpc/1.0.0.zip: no such file or directory
 ```
If I revert back to Terragrunt 0.53 then things works again.

Any help, what could be going wrong here? 

---

<ins datetime="2024-10-05T11:25:30Z">
  <p><a href="https://support.gruntwork.io/hc/requests/111845">Tracked in ticket #111845</a></p>
</ins>

**Comment by denis256** [comment]
Hi,
Sharing more details about your environment would be helpful (OS, Terragrunt installation method, Terraform version). 
Consider creating an example repo that reproduces the issue.

I did a quick test in my environment (Ubuntu 22.04, x86_64, Terragrunt 0.55.6 from GitHub release) and it works without the mentioned issue:
```
$ terragrunt.0.55.6 init
INFO[0000] Downloading Terraform configurations from https://my.artifactory-devops.com/artifactory/my-custom-modules-terraform-local/aws/my-vpc/1.0.0.zip into /projects/gruntwork/terragrunt-tests/discussion-844/.terragrunt-cache/1nvu5L9gbzbYkx5M5hZZ6a7FGAo/i78rPMG9EaJXhFyYmO9xCFmi_Vg 
...
```

You can find my test setup here: https://github.com/denis256/terragrunt-tests/tree/master/discussion-844
